q2cli/builtin/tools.py

```diff
@@ -195,7 +195,89 @@ def peek(path):
         click.echo(metadata.format)
 
 
-@tools.command('inspect-metadata',
+@tools.command(name='cast-metadata',
+               short_help='Designate metadata columns as numerical'
+                          ' or categorical.',
+               help='Designating the column types within a metadata file.'
+                    ' Supported column types are numerical or categorical.'
+                    ' Providing multiple file paths to this command will merge'
+                    ' the metadata.',
+               cls=ToolCommand)
+# NOTE: # This gives a potential way to avoid multiple --cast calls but it's
+# fiddely https://stackoverflow.com/a/47730333
+@click.option('--cast', required=True, multiple=True,
+              help='Cast flags for each metadata column that should be'
+              ' specified as either categorical or numeric column types.')
+@click.option('--error-on-extra', is_flag=True,
+              help='Cast flags provided inline that do not exist within the'
+              ' metadata provided will result in a raised error.'
+              ' Enabled by default. If this flag is disabled and'
+              ' extra flags are provided, they will be ignored.')
+@click.option('--ignore-missing', is_flag=True,
+              help='Cast flags not provided inline that exist within the'
+              ' metadata provided will be ignored. Enabled by default.'
+              ' If this flag is disabled and not all column flags'
+              ' are provided, an error will be raised.')
+@click.option('--output-file', required=True,
+              type=click.Path(exists=False, file_okay=True, dir_okay=True,
```

Comment:
> dir_okay=True

There appears to be a minor logic issue here:

```bash
(q2dev)
~/projects/qiime2/q2cli
☻  mkdir md1out                                                                                                cast_metadata_tool 6ccc7c7 ✗
(q2dev)
~/projects/qiime2/q2cli
☻  qiime tools cast-metadata md1.tsv --cast foo:categorical --output-file md1out/                              cast_metadata_tool 6ccc7c7 ✗
Traceback (most recent call last):
  File "/Users/matthew/.conda/envs/q2dev/bin/qiime", line 33, in <module>
    sys.exit(load_entry_point('q2cli', 'console_scripts', 'qiime')())
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/matthew/projects/qiime2/q2cli/q2cli/builtin/tools.py", line 277, in cast_metadata
    cast_md.save(output_file)
  File "/Users/matthew/projects/qiime2/qiime2/qiime2/metadata/metadata.py", line 222, in save
    MetadataWriter(self).write(filepath)
  File "/Users/matthew/projects/qiime2/qiime2/qiime2/metadata/io.py", line 346, in write
    with open(filepath, 'w', newline='', encoding='utf-8') as fh:
IsADirectoryError: [Errno 21] Is a directory: 'md1out/'
```

Reply:
Good catch @thermokarst, this should be set to false. Will update this!